src/github-api-helper.ts

```diff
@@ -125,19 +125,20 @@ async function downloadArchive(
   commit: string,
   baseUrl?: string
 ): Promise<Buffer> {
-  const octokit = getOctokit(authToken, {baseUrl: baseUrl})
-  const params: Octokit.ReposGetArchiveLinkParams = {
+  const octokit = github.getOctokit(authToken, {baseUrl: baseUrl})
+  const download = IS_WINDOWS
+    ? octokit.rest.repos.downloadZipballArchive
+    : octokit.rest.repos.downloadTarballArchive
+  const response = await download({
     owner: owner,
     repo: repo,
-    archive_format: IS_WINDOWS ? 'zipball' : 'tarball',
     ref: commit || ref
-  }
-  const response = await octokit.repos.getArchiveLink(params)
+  })
   if (response.status != 200) {
```

Comment:
OK I need help on how to properly detect error from the status code.
```js
if (response.status != 200) {
```
causes a compile error saying the comparison is always false.

Reply:
```js
if (response.status != 302) {
```
does not work either because its value is 200 not 302 in a successful request.

Reply:
At a glance, it looks like a 302 response code is guaranteed at this point, implying that `octokit.rest.repos.downloadZip|TarballArchive` would throw in case it fails.

So we don't need to explicitly check the response code here anymore IMO, `download` will throw and it'll propagate up.